core/src/main/java/org/apache/calcite/util/javac/CalciteCompilerArgs.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.util.javac;
+
+/**
+ * A <code>CalciteCompilerArgs</code> holds the arguments for a
+ * {@link org.codehaus.janino.SimpleCompiler} and its sub classes.
+ *
+ */
+public class CalciteCompilerArgs {
+
+  public static final CalciteCompilerArgs DEFAULT = new CalciteCompilerArgs();
+
+  private ClassLoader classLoader = CalciteCompilerArgs.class.getClassLoader();
+
+  public void setClassLoader(ClassLoader classLoader) {
+    if (classLoader != null) {
+      this.classLoader = classLoader;
+    }
```

Comment:
Please, no. "shared mutable state" is not how Calcite is configured.
It looks like the classloader should be an extra argument to `RexExecutable` constructor or something like that